File: GptFactChecker/GptFactCheckerApi/Model/Source.cs
namespace GptFactCheckerApi.Model;

public class Source
{
    public string Id { get; set; }
    public string Name { get; set; }
    public string Language { get; set; }
    public string Description { get; set; }
    public List<string> Tags { get; set; }

    public string SourceType { get; set; }
    public string SourcePerson { get; set; }
    public string SourceContext { get; set; }
    public string SourceUrl { get; set; } 

    public DateOnly SourceImportedDate { get; set; }
    public DateOnly SourceCreatedDate { get; set; }

    public DateOnly ClaimsFirstExtractedDate { get; set; }
    public DateOnly ClaimsUpdatedDate { get; set; }
   
    public string CoverImageUrl { get; set; }
}

File: GptFactChecker/gpt-fact-checker-front/src/model/Source.ts
import type Claim from "./Claim";

class Source {
	public id: string;
	public name: string;
	public language: string;
	public description: string;
	public coverImageUrl: string;

	public tags: string[];
	
	public sourceType: string;
	public sourcePerson: string;
	public sourceContext: string;
	public sourceUrl: string;
	public sourceRawText: string;
	public sourceImportedDate?: string;
	public sourceCreatedDate?: string;

	public claimsFirstExtractedDate: string;
	public claimsUpdatedDate: string;
	public claims?: Claim[];

	constructor() {
		this.id = "";
		this.name = "";
		this.language = "";
		this.description = "";
		this.tags = [];
		this.sourceType = "";
		this.sourcePerson = "";
		this.sourceContext = "";
		this.sourceUrl = "";
		this.sourceImportedDate = "";
		this.sourceCreatedDate = "";
		this.claimsFirstExtractedDate = "";
		this.claimsUpdatedDate = "";
		this.coverImageUrl = "";
		this.sourceRawText = "";
	}
}

export default Source;
